pallets/parachain-staking/src/lib.rs

```diff
@@ -2284,6 +2324,20 @@ pub mod pallet {
 			// prepare unstaking of collator candidate
 			Self::prep_unstake_exit_queue(&state.id, state.stake);
 
+			// disable validator for next session if they were in the set of validators
+			if let Some(index) = pallet_session::Pallet::<T>::validators()
```

Comment:
I would maybe not use the `let if` here but rather add `.map(pallet_session::Pallet::<T>::disable_index)` to the end of the chain

Reply:
Thanks, much more elegant. Fixed in e547d10